sidecar/supervisor/appsupervisor.go

```diff
@@ -94,8 +103,12 @@ func (a *AppSupervisor) DoAppSupervision(cmdArgs []string, agent *register.Regis
 func (a *AppSupervisor) Shutdown(exitCode int) {
 	// TODO: Gracefully shutdown Ngnix, and filebeat
 	//ugly temporary hack to kill all processes in container
-	exec.Command("pkill", "-9", "-f", "nginx")
-	exec.Command("pkill", "-9", "-f", "filebeat")
+	//exec.Command("pkill", "-9", "-f", "nginx")
+	//exec.Command("pkill", "-9", "-f", "filebeat")
+
+	for _, cmd := range a.cmds {
+		cmd.Process.Kill()
+	}
 
 	if a.agent != nil {
 		a.agent.Stop()
```

Comment:
There's also several items missing from this change-set:
1. Need to properly manage Nginx process(es).
2. Need to provide configuration option for multiple processes (via YAML file).
3. Given that the sidecar is expected to run as [PID 1](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/) in the container, we need to properly reap adopted child processes.